### DISCUSSION TITLE
How to use rich to render exceptions with markdown and/or mixed data?

### DISCUSSION THREAD

**Discussion by kratsg** [Textualize/rich#2684] [General]
```python
from rich.console import Console, Group
from rich.markdown import Markdown
from rich.Text import Text

from rich.traceback import install
install()

raise Exception(Group(Text('some text here'), Markdown('and some **bolded** text here')))
```

similarly, a simpler version of this also doesn't quite work as I expect

```python
from rich.markdown import Markdown

from rich.traceback import install
install()

raise Exception(Markdown('and some **bolded** text here'))
```

Am I thinking about this the wrong way or am I missing something in the documentation?

**Comment by TechWiz-3** [comment]
Maybe open a discussion for this instead of an issue

**Comment by willmcgugan** [comment]
Exceptions generally take strings. Rich doesn't make any assumptions beyond that.

**Reply by anatoly-scherbakov** [reply]
I have written a little library called [documented](https://github.com/anatoly-scherbakov/documented) which overrides `__str__` method of an exception class to output the exception's docstring instead of the default text. (It also can fill placeholders and thus make the exception rendition dynamic.)

I tried to use the default exception hook and `__rich_repr__` protocol to interpet console and/or Markdown markup in the exception's `__str__` but to no avail. `[red]something[/red]` or `*something*` are not being interpreted.

Would be happy for a direction to look into. Thanks!
